File: domain/repository/bundle.go
package repository

import (
	"rizhua.com/application/query"
	"rizhua.com/infrastructure/persistence/po"
)

type Bundle interface {
	// 创建套餐
	Create(po.Bundle) error

	// 删除套餐
	Delete(id []int64) error

	// 更新套餐
	Update(po.Bundle) error

	// 套餐详情
	Get(query.Request) (po.Bundle, error)

	// 套餐列表
	Find(query.Request) ([]po.Bundle, int64, error)

	// 套餐节点 id 列表
	FindNodeID(bundleID int64) ([]po.BundleNode, error)

	// 绑定、解绑节点
	BindNodeID(bundleID int64, nodeID []int64) error

	// 套餐节点列表
	FindNode(bundleID int64, path string) ([]po.Node, error)

	// 许可详情
	GetLicense(query.Request) (po.License, error)

	// 许可列表
	FindLicense(query.Request) ([]po.License, int64, error)
}

File: web/packages/adm/src/service/api/bundle.ts
import { model, http } from "../";

export class Bundle {
    static create(data: model.Bundle): Promise<model.Response> {
        let url = '/bundle/create';
        return http.post(url, data);
    }

    static delete(data: {id: number[]}): Promise<model.Response> {
        let url = '/bundle/delete';
        return http.post(url, data);
    }

    static update(data: model.Bundle): Promise<model.Response> {
        let url = '/bundle/update';
        return http.post(url, data);
    }

    static find(data: model.Request): Promise<model.Response> {
        let url = '/bundle/find';
        return http.post(url, data);
    }

    static bindNodeId(data: {bundleId: number, nodeId: number[]}): Promise<model.Response> {
        let url = '/bundle/bindNodeId';
        return http.post(url, data);
    }

    static findNodeId(data: {id: number}): Promise<model.Response> {
        let url = '/bundle/findNodeId';
        return http.post(url, data);
    }

    static findLicense(data: model.Request): Promise<model.Response> {
        let url = '/bundle/license/find';
        return http.post(url, data);
    }
}